samples/csharp/MicrosoftEduGraphSamples/Workflows/AssignmentWorkflow.cs

```diff
@@ -64,21 +64,27 @@ public async Task<IEnumerable<EducationAssignment>> GetMeAssignmentsFromNonArchi
         {
             try
             {
-                //Check user details for assignments               
+                //Check user details for assignments
                 string userAccount = isTeacher ? _config["teacherAccount"] : _config["studentAccount"];
 
                 // Get a Graph client using delegated permissions
-                var graphClient = MicrosoftGraphSDK.GraphClient.GetDelegateClient(_config["tenantId"], _config["appId"], userAccount, _config["password"]);
+                var graphClient = GraphClient.GetDelegateClient(_config["tenantId"], _config["appId"], userAccount, _config["password"]);
 
-                // Call to get user classes
-                var joinedTeams = await graphClient.GetJoinedTeamsAsync();
+                // Get a Graph client using app-only permissions
+                var graphAppClient = GraphClient.GetApplicationClient(_config["tenantId"], _config["appId"], _config["secret"]);
+
+                // Get user info
+                var teacher = await MicrosoftGraphSDK.User.GetUserInfoAsync(graphClient);
+
+                // Call to get user joined teams
+                var joinedTeams = await graphAppClient.GetUserJoinedTeamsAsync(teacher.Id);
```

Comment:
can we still use delegate for this instead of creating an AppOnly client? I think that would be preferrable (unless the delegate auth has been deprecated)

Reply:
done, the app registration was missing the delagted permissions for this call